AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -121,6 +119,8 @@ - (id)init {
         self.capturingContext = malloc(sizeof(AudioCapturerContext));
         memset(self.capturingContext, 0, sizeof(AudioCapturerContext));
         self.capturingContext->bufferList = &_captureBufferList;
+        
+        [self setupAVAudioSession];
```

Comment:
I ran into the following crash while testing the changes on AVAudioChangeExample, made the fix the same PR -

![Screen Shot 2020-01-15 at 3 45 25 PM](https://user-images.githubusercontent.com/17037593/72481630-f9531e80-37af-11ea-8e7e-5f73b22ff459.png)

Reply:
Right, the route change happens on a background thread and init happens in the main thread. we were assigning AVAudioSession callbacks priorto setting the context in init, so it is possible that AVAudioSession gives you callback before we finish init.